meta.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zap
+
+import (
+	"os"
+
+	"github.com/uber-go/atomic"
+)
+
+// Meta is implementation-agnostic state management for Loggers. Most Logger
+// implementations can reduce the required boilerplate by embedding a *Meta.
+//
+// Note that while the level-related fields and methods are safe for concurrent
+// use, the remaining fields are not.
+//
+// TODO: Consider better names for this before releasing 1.0.
+type Meta struct {
+	Development bool
+	Encoder     encoder
+	Hooks       []hook
+	Output      WriteSyncer
+	ErrorOutput WriteSyncer
+
+	lvl *atomic.Int32
+}
+
+// NewMeta returns a new meta struct with sensible defaults: logging at
+// InfoLevel, a JSON encoder, development mode off, and writing to standard error
+// and standard out.
+func NewMeta() *Meta {
```

Comment:
It's still better to embed it by value since e.g. jsonLogger is already always boxed. In other words, embedding by value into a struct that is already allocated on the heap saves an allocation over embedding by pointer.